### PR TITLE
[WIP] [Feature] instance-initializer blueprint for module unification

### PR DESCRIPTION
**Pull request by lbeauvilliers**
To Do:
- [x] Get the import statements generating correctly for the in-app case
- [ ] Write tests for the in-addon case
- [ ] Update `instance-initializer-test-test.js `

### CODE REVIEW

File: .idea/ember.js.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/.tmp" />
+      <excludeFolder url="file://$MODULE_DIR$/temp" />
+      <excludeFolder url="file://$MODULE_DIR$/tmp" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (GavinJoyce) [inline .idea/ember.js.iml]: can you :zap: these files (and perhaps add `.idea/` to `.gitignore`)?

File: node-tests/blueprints/instance-initializer-test.js [hunk 2]
```diff
@@ -116,6 +118,59 @@ describe('Blueprint: instance-initializer', function() {
...
+          "  // appInstance.inject('route', 'foo', 'service:foo');\n" +
+          '}\n' +
+          '\n' +
+          'export default {\n' +
+          '  initialize\n' +
+          '};'
+        );
+
+        expect(_file('src/instance-initializers/foo-test.js')).to.contain(
+          "import { initialize } from 'my-app/instance-initializers/foo';"
```
Comment (GavinJoyce) [inline node-tests/blueprints/instance-initializer-test.js]: I think this will have to change to point to the new path under src

**Comment by GavinJoyce** [comment]
We'll also need to add tests to [instance-initializer-test-test.js](https://github.com/emberjs/ember.js/blob/master/node-tests/blueprints/instance-initializer-test-test.js) to ensure that we're generating co-located tests. 

I added a similar comment [here](https://github.com/emberjs/ember.js/pull/16489#issuecomment-381505216), it looks like we'll have to make a change in https://github.com/ember-cli/ember-load-initializers to allow tests to be co-located

**Comment by GavinJoyce** [comment]
@lbeauvilliers  ^ that's been resolved, perhaps you could rebase?

Also, could you add an in-repo-addon test [similar to this](https://github.com/emberjs/ember.js/pull/16553)? This will ensure that we're not breaking existing functionality.

LMK if you'd like to pair on this sometime

File: node-tests/blueprints/instance-initializer-test.js (hunk 2 above)
Comment (lbeauvilliers) [reply node-tests/blueprints/instance-initializer-test.js]: Thanks @GavinJoyce, changed that--now trying to figure out how to get the tests actually passing!

When I have this worked out, I'll move onto the `addon` case, then update `instance-initializer-test-test`.

**Comment by GavinJoyce** [comment]
If anyone wants to pick this PR up, please do so

**Comment by ppcano** [comment]
Closed by #17235

**Pull request closed without merging**